src/main/java/com/charlesgadeken/entwined/EntwinedTriggers.java

```diff
@@ -0,0 +1,390 @@
+                    blurEffect.level,
+                    speedEffect.speed,
+                    spinEffect.spin,
+                    candyCloudTextureEffect.amount
+                };
+    }
+
+    public void registerPattern(EntwinedBasePattern pattern, String nfcSerialNumber) {
+        registerPattern(pattern, nfcSerialNumber, 2);
+    }
+
+    public void registerPattern(
+            EntwinedTriggerablePattern pattern, String nfcSerialNumber, int apc40DrumpadRow) {
+        registerVisual(pattern, nfcSerialNumber, apc40DrumpadRow, VisualType.Pattern);
+    }
+
+    void registerVisual(
+            EntwinedTriggerablePattern pattern,
+            String nfcSerialNumber,
+            int apc40DrumpadRow,
+            VisualType visualType) {
+
+        LXBlend t = new DissolveBlend(lx);
+
+        // NOTE(meawoppl) @Slee same question below.  re `.setDuration(dissolveTime);`
+        pattern.setTransition(t);
+
+        Triggerable triggerable = configurePatternAsTriggerable(pattern);
+        BooleanParameter toggle = null;
+        if (apc40Drumpad != null) {
+            toggle =
+                    apc40DrumpadTriggerablesLists[apc40DrumpadRow].size() < 9
+                            ? nfcToggles[apc40DrumpadRow][
+                                    apc40DrumpadTriggerablesLists[apc40DrumpadRow].size()]
+                            : null;
+            apc40DrumpadTriggerablesLists[apc40DrumpadRow].add(triggerable);
+        }
+        if (nfcEngine != null) {
+            nfcEngine.registerTriggerable(nfcSerialNumber, triggerable, visualType, toggle);
+        }
+    }
+
+    Triggerable configurePatternAsTriggerable(EntwinedTriggerablePattern pattern) {
+        LXChannel channel = lx.engine.mixer.addChannel(new EntwinedBasePattern[] {pattern});
+        setupChannel(channel, false);
+
+        pattern.onTriggerableModeEnabled();
+        return pattern.getTriggerable();
+    }
+
+    void setupChannel(final LXChannel channel, boolean noOpWhenNotRunning) {
+        // @Slee, honestly no idea what the intention is here...
+        // It looks like patterns used to have associated traditionss" have have moved to the
+        // `blend` language, but I don't seen any analogue for that remaining....
+        channel.setFaderTransition(
+                new TreesTransition(lx, channel, model, channelTreeLevels, channelShrubLevels));
+
+        // channel.transitionBlendMode.setObjects(new TreesTransition[] {new TreesTransition(lx, channel, model, channelTreeLevels, channelShrubLevels)})
+        // ??
+
+
+        channel.addListener(
+                new LXChannel.AbstractListener() {
+                    LXTransition transition;
+
+                    @Override
+                    public void patternWillChange(
+                            LXChannel channel, LXPattern pattern, LXPattern nextPattern) {
+                        if (!channel.enabled.isOn()) {
+                            transition = nextPattern.getTransition();
```

Comment:
Per the above comment about per-pattern transitions going away, this is maybe just all not relevant anymore. I'd just pick a sensible transition to assign at the channel level and leave it.